tools/lint-md/lint-md.src.mjs

```diff
@@ -28,11 +28,24 @@ const linter = unified()
 
 paths.forEach(async (path) => {
   const file = await read(path);
+  // We need to calculate `fileContents` before running `linter.process(files)`
+  // because `linter.process(files)` mutates `file` and returns it as `result`.
+  // So we won't be able to use `file` after that to see if its contents have
+  // changed as they will have been altered to the changed version.
+  const fileContents = file.toString();
   const result = await linter.process(file);
-  if (format) {
+  const isDifferent = fileContents !== result.toString();
+  if (format && isDifferent) {
     fs.writeFileSync(path, result.toString());
```

Comment:
Shouldn't it be a sub-condition? (also do we need the `toString()` call?)
```suggestion
  if (format) {
    if (isDifferent) fs.writeFileSync(path, result);
```

Reply:
Whoops, yeah, that's what I get for early a.m. coding on a Saturday.

Reply:
We need to use `toString()`. Otherwise this happens:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of VFile
    at Object.writeFileSync (node:fs:2146:5)
    at file:///Users/trott/io.js/tools/lint-md/lint-md.src.mjs:40:10 {
  code: 'ERR_INVALID_ARG_TYPE'
}
```